### PR TITLE
MOTOR-1059 Single Source the Package Version

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Missing the setup.py changes?

**Comment by blink1073** [comment]
> Missing the setup.py changes?

Fixed

File: motor/_version.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Version-related data for motor."""
+
+version_tuple = (3, 1, 2, ".dev0")
```
Comment (ShaneHarvey) [inline motor/_version.py:17]: `".dev0"` -> `"dev0"`

otherwise version will be '3.1.2..dev0'

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: motor/_version.py (hunk 1 above)
Comment (ShaneHarvey) [reply motor/_version.py:17]: Or we could fix get_version_string to avoid that by removing leading `.`'s or something.

Comment (blink1073) [reply motor/_version.py:17]: I updated it to account for "rc" releases.

Comment (ShaneHarvey) [reply motor/_version.py:17]: How bout we copy what pymongo does to be consistent: https://github.com/mongodb/mongo-python-driver/blob/1d117c1f39983ec88376816c7e3fd73fb140b863/pymongo/_version.py#L21-L24

Pymongo's version is more robust, eg it handles `version_tuple = (3, 1, ".dev0")`

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

**Comment by blink1073** [comment]
> How bout we copy what pymongo does to be consistent

Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**